src/App.tsx

```diff
@@ -15,48 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const adoptedGoods = goodsFromServer.map((good, id) => ({ name: good, id }));
```

Comment:
```suggestion
const adoptedGoods = goodsFromServer.map(({ good, id }) => ({ name: good, id }));
```
your code will not took the id from the good, it will use index as it is second param in map callback function